core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PreferWritePartitioning.java

```diff
@@ -46,6 +49,28 @@ public Result apply(TableWriterNode node, Captures captures, Context context)
             return Result.empty();
         }
 
+        int minimumNumberOfPartitions = getPreferredWritePartitioningMinNumberOfPartitions(context.getSession());
+        if (minimumNumberOfPartitions <= 1) {
+            return enable(node);
+        }
+
+        double expectedNumberOfPartitions = getRowsCount(
+                context.getStatsProvider().getStats(node.getSource()),
+                node.getPreferredPartitioningScheme().get().getPartitioning().getColumns());
+
+        if (isNaN(expectedNumberOfPartitions)) {
```

Comment:
make a single iff of `isNaN(expectedNumberOfPartitions)` and `expectedNumberOfPartitions < minimumNumberOfPartitions`